### PR TITLE
build(docker): metadata-ingestion images build and add slim version

### PR DESCRIPTION
**Pull request by david-leifker**
- Update metadata-ingestion dockerize
- Split `base` and `full` metadata-ingestion images (the latter was not renamed)
- Add `slim` metadata-ingestion w/o pyspark

The gradle changes just allowed me to test the dockerfiles easily.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: .github/workflows/docker-ingestion-base.yml [hunk 1]
```diff
@@ -40,38 +40,6 @@ jobs:
           username: ${{ secrets.ACRYL_DOCKER_USERNAME }}
           password: ${{ secrets.ACRYL_DOCKER_PASSWORD }}
           publish: ${{ github.ref == 'refs/heads/master' }}
-          context: ./docker/datahub-ingestion
-          file: ./docker/datahub-ingestion/base.Dockerfile
+          context: .
+          file: ./docker/datahub-ingestion-base/Dockerfile
           platforms: linux/amd64,linux/arm64
-  ingestion-base_scan:
```
Comment (david-leifker) [inline .github/workflows/docker-ingestion-base.yml:46]: Moved the scan to the `slim` image

File: docker/datahub-ingestion-slim/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Defining environment
+ARG APP_ENV=prod
+ARG DOCKER_VERSION=latest
+
+FROM acryldata/datahub-ingestion:$DOCKER_VERSION as base
+
+USER 0
+RUN pip uninstall -y pyspark
```
Comment (david-leifker) [inline docker/datahub-ingestion-slim/Dockerfile:8]: Only change for `slim` is removing pyspark at the moment.

File: docker/datahub-ingestion/Dockerfile [hunk 3]
```diff
@@ -1,10 +1,12 @@
 # Defining environment
 ARG APP_ENV=prod
+ARG DOCKER_VERSION=latest
 
-FROM acryldata/datahub-ingestion-base as base
+FROM acryldata/datahub-ingestion-base:$DOCKER_VERSION as base
 
-FROM openjdk:11 as prod-build
+FROM eclipse-temurin:11 as prod-build
```
Comment (david-leifker) [inline docker/datahub-ingestion/Dockerfile:7]: The openjdk image was abandoned last year. This is the recommended replacement. This is a build container so not part of the final container.

File: .github/workflows/docker-ingestion.yml [hunk 4]
```diff
@@ -91,3 +92,57 @@ jobs:
...
+      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
+      actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
+    name: "[Monitoring] Scan datahub-ingestion-slim images for vulnerabilities"
+    if: ${{ github.ref == 'refs/heads/master' }}
+    runs-on: ubuntu-latest
+    needs: [push_to_registries]
+    steps:
+      - name: Checkout # adding checkout step just to make trivy upload happy
+        uses: actions/checkout@v3
+      - name: Download image
```
Comment (david-leifker) [inline .github/workflows/docker-ingestion.yml:129]: This required the use of the custom action from the unified workflow. Using this unified workflow custom action in order to hopefully make this work (unlike [this](https://github.com/datahub-project/datahub/pull/7401)).

File: .github/workflows/docker-ingestion.yml [hunk 5]
```diff
@@ -91,3 +92,57 @@ jobs:
...
+        uses: crazy-max/ghaction-docker-meta@v1
+        with:
+          # list of Docker images to use as base name for tags
+          images: |
+            linkedin/datahub-ingestion-slim
+          # add git short SHA as Docker tag
+          tag-custom: ${{ needs.setup.outputs.tag }}
+          tag-custom-only: true
+      - name: Build and Push image (slim)
+        uses: ./.github/actions/docker-custom-build-and-push
```
Comment (david-leifker) [inline .github/workflows/docker-ingestion.yml:106]: This seems to be required for scans, see comment below.

**Review by RyanHolstien — APPROVED** [review]

File: docker/datahub-ingestion/Dockerfile (hunk 3 above)
Comment (david-leifker) [reply docker/datahub-ingestion/Dockerfile:7]: This is the only place we use an image like this. In other places we are using alpine based images.

**Comment by david-leifker** [comment]
smoke test errors are not related to ingestion containers, broken master tests are the cause. backlog of jobs is very high, bypassing the smoke test errors.

**Pull request merged**